api/v1/db/awsS3/aws-operations.js

```diff
@@ -180,18 +179,17 @@ const updateMetadata = async (metadata, key, bucket = thisBucket) => {
  * @param {string} bucket The bucket where the object is located
  * @returns {Promise} Promise object representing the response. undefined if the key was not found
  */
-const deleteObject = (key, bucket = thisBucket) => new Promise((resolve, reject) => {
+const deleteObject = async (key, bucket = thisBucket) => {
   const params = { Bucket: bucket, Key: key };
-  s3.deleteObject(params).promise().then((data) => {
-    resolve(data);
-  }).catch((err) => {
+  try {
+    return await s3.deleteObject(params).promise();
+  } catch (err) {
     if (err.code === 'NotFound') {
-      resolve(undefined);
-    } else {
-      reject(err);
+      return undefined;
     }
-  });
-});
+    throw err;
+  }
+};
```

Comment:
This one is also very similar. One idea is like what I did in students API [here](https://github.com/osu-mist/students-api/blob/master/api/v1/db/oracledb/students-dao.js#L19-L53) to create a generic function and make the different components like function calls as parameters. What do you think?